keras/backend/numpy/numpy.py

```diff
@@ -2,6 +2,7 @@
 
 from keras.backend import config
 from keras.backend import standardize_dtype
+from keras.backend.common import dtypes
 
 
 def add(x1, x2):
```

Comment:
Can you try adding `result_dtype` to an op that has 2 tensor inputs, like `add`?

Reply:
I have added `result_type` for `ops.add` and it works as expected.

A new option `pre_canonicalize` for `result_type` is added to align the behavior with JAX.